tests/core/search/autocompleteresponsetransformer.js

```diff
@@ -107,4 +107,143 @@ describe('transform autocomplete response', () => {
       AutoCompleteResponseTransformer.transformFilterAutoCompleteResponse(responseFromCore);
     expect(actualTransformedResponse).toEqual(expectedTransformedResponse);
   });
+
+  it('transform filter autocomplete response without sections', () => {
+    const responseFromCore = {
+      sectioned: false,
+      sections: [],
+      results: [
+        {
+          value: 'Virginia Beach',
+          matchedSubstrings: [
+            {
+              offset: 0,
+              length: 8
+            }
+          ],
+          filter: {
+            comparator: '$eq',
+            comparedValue: 'Virginia Beach',
+            fieldId: 'name'
+          },
+          key: 'name'
+        }
+      ],
+      inputIntents: [],
+      queryId: '42d5b709-3b9f-464a-b9b5-764467cbf540'
+    };
+    const expectedTransformedResponse = {
+      inputIntents: [],
+      queryId: '42d5b709-3b9f-464a-b9b5-764467cbf540',
+      sections: [
+        {
+          results: [
+            {
+              filter: {
+                name: {
+                  $eq: 'Virginia Beach'
+                }
+              },
+              intents: [],
+              key: 'name',
+              matchedSubstrings: [
+                {
+                  length: 8,
+                  offset: 0
+                }
+              ],
+              shortValue: 'Virginia Beach',
+              value: 'Virginia Beach'
+            }
+          ],
+          resultsCount: 1
+        }
+      ]
+    };
+    const actualTransformedResponse =
+      AutoCompleteResponseTransformer.transformFilterAutoCompleteResponse(responseFromCore);
+    expect(actualTransformedResponse).toEqual(expectedTransformedResponse);
+  });
+
+  it('transform filter', () => {
+    const filterFromCore = {
+      comparator: '$eq',
+      comparedValue: 'Arlington',
+      fieldId: 'c_name'
+    };
+    const expectedTransformedFilter = {
+      c_name: {
+        $eq: 'Arlington'
+      }
+    };
+    const actualTransformedFilter =
+      AutoCompleteResponseTransformer._transformFilter(filterFromCore);
+    expect(actualTransformedFilter).toEqual(expectedTransformedFilter);
+  });
+
+  it('transform autocomplete result with filter', () => {
+    const resultFromCore = {
+      value: 'Virginia Beach',
+      matchedSubstrings: [
+        {
+          offset: 0,
+          length: 8
+        }
+      ],
+      filter: {
+        comparator: '$eq',
+        comparedValue: 'Virginia Beach',
+        fieldId: 'name'
+      },
+      key: 'name'
+    };
+    const expectedTransformedResult = {
+      filter: {
+        name: {
+          $eq: 'Virginia Beach'
+        }
+      },
+      intents: [],
+      key: 'name',
+      matchedSubstrings: [
+        {
+          length: 8,
+          offset: 0
+        }
+      ],
+      shortValue: 'Virginia Beach',
```

Comment:
This should be removed since we are removing shortValue, right?

Reply:
Ah, not quite. The `AutoCompleteResult` of the SDK itself actually has `shortValue` as a property, and it is used elsewhere in the SDK, so for the transformed result, we should be expecting `shortValue`.

Reply:
ah, got it